exporters/otlp/include/opentelemetry/exporters/otlp/otlp_http_exporter.h

```diff
@@ -0,0 +1,107 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+// We need include exporter.h first, which will include Windows.h with NOMINMAX on Windows
+#include "opentelemetry/sdk/trace/exporter.h"
+
+#include <memory>
+#include <mutex>
+#include <string>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace otlp
+{
+// The default URL path to post trace data.
+constexpr char kDefaultTracePath[] = "/v1/traces";
+// The default URL path to post metric data.
+constexpr char kDefaultMetricsPath[] = "/v1/metrics";
+// The default URL path to post metric data.
+constexpr char kDefaultLogPath[] = "/v1/logs";
+// The HTTP header "Content-Type"
+constexpr char kHttpContentType[] = "application/json";
+
+enum class BytesMappingKind
+{
+  kHexId,
+  kBase64,
+  kHex,
+};
+
+/**
+ * Struct to hold OTLP exporter options.
+ */
+struct OtlpHttpExporterOptions
+{
+  // The endpoint to export to. By default
+  // @see
+  // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/otlp.md
+  // @see https://github.com/open-telemetry/opentelemetry-collector/tree/main/receiver/otlpreceiver
+  std::string url = std::string("http://localhost:4317") + kDefaultTracePath;
+
+  // TODO: By default when false, set CURLOPT_SSL_VERIFYPEER to false
+  // bool use_ssl_credentials = false;
+  // If convert bytes into hex. By default, we will convert bytes into base64
+  BytesMappingKind json_bytes_mapping = BytesMappingKind::kHexId;
+
+  // If using the json name of protobuf field to set the key of json. By default, we will use the
+  // field name just like proto files.
+  bool use_json_name = false;
+
+  // Whether to print the status of the exporter in the console
+  bool console_debug = false;
+
+  // Maximum time to wait for response after sending http request(milliseconds)
+  int response_timeout = 30000;
+};
+
+/**
+ * The OTLP exporter exports span data in OpenTelemetry Protocol (OTLP) format.
+ */
+class OtlpHttpExporter final : public opentelemetry::sdk::trace::SpanExporter
+{
+public:
+  /**
+   * Create an OtlpHttpExporter using all default options.
+   */
+  OtlpHttpExporter();
+
+  /**
+   * Create an OtlpHttpExporter using the given options.
+   */
+  OtlpHttpExporter(const OtlpHttpExporterOptions &options);
```

Comment:
Nit - can we make this `explicit`. I think this is missing for `OtlpGrpcExporter` too which also should be modified.

Reply:
My negligence. I will add `explicit` later and add some tests.